labs/lab-2-javascript-game/script.js

```diff
@@ -0,0 +1,50 @@
+function startGame() {
+  const play = window.prompt("Do you want to play?");
+  if (play === "yes") {
+    var name = window.prompt("Enter your name:")
+    startCombat(name);
+  }
+}
+
+function getDamage() {
+  const min = 1;
+  const max = 5;
+  return (Math.floor(Math.random() * (max - min + 1) + min));
+}
+
+function startCombat(name) {
+  var winCount = 0;
+  var userHP = 40;
+  var grantHP = 10;
+  var choice;
+
+  while ((winCount < 3) && (userHP > 0)) {
+    choice = window.prompt("Would you like to attack or quit?");
+    if (choice === "attack") {
+      userHP = userHP - getDamage();
+      grantHP = grantHP - getDamage();
+      console.log(name + " has " + userHP + " health left.");
+      console.log("Grant the Mighty Chicken has " + grantHP + " health left.");
+    } else if (choice === "quit") {
+      winCount = 4;
+    }
+    if (grantHP <= 0) {
+      winCount++;
+      grantHP = 10;
+      if (winCount !== 3) { // Note: Does not show if game has finished.
+        console.log("You beat Grant! You need to win " + (3 - winCount) + " more round(s).");
+      }
+    }
+  }
+
+  // Battle results
```

Comment:
What if you broke out a function for reporting results, which you handed a winCount and whether or not the user quit the game?